lib/matplotlib/colorbar.py

```diff
@@ -244,6 +244,73 @@ def tick_values(self, vmin, vmax):
         return ticks[(ticks >= vmin) & (ticks <= vmax)]
 
 
+class _ColorbarAutoMinorLocator(ticker.AutoMinorLocator):
+    """
+    AutoMinorLocator for Colorbar
+
+    This locator is just a `.AutoMinorLocator` except the min and max are
+    clipped by the norm's min and max (i.e. vmin/vmax from the
+    image/pcolor/contour object).  This is necessary so that the minorticks
+    don't extrude into the "extend regions".
+    """
+    def __init__(self, colorbar, n=None):
+        """
+        This ticker needs to know the *colorbar* so that it can access
+        its *vmin* and *vmax*.
+        """
+
+        self._colorbar = colorbar
+        self.vmin = self._colorbar.norm.vmin
+        self.vmax = self._colorbar.norm.vmax
+        self.ndivs = n
+
+    def __call__(self):
```

Comment:
Can't you just call `ticker.AutoMinorLocator` and then trim the values?

Reply:
ticker.AutoMinorLocator doesn't seem to have a tick_values method that would return the tick locations.

Reply:
Thanks @jklymak  I have pushed a  cleaner version now.